### PR TITLE
support for django-rest-framework 3

### PR DESCRIPTION
**Pull request by smcoll**
Take a careful look at the tests; DRF3 requires explicit querysets, and complains about naming 'source' if the source is the same as the field name, so i had to modify the tests a bit.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 version = '0.1.0'
 
 install_requires = [
-    'djangorestframework>=2.3.14,<3',
+    'djangorestframework>=2.3.14',
```
Comment (Ian-Foote) [inline setup.py]: Better to require:
```python
'djangorestframework>=3.0.0,<4'
```

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ env:
   - DJANGO_VERSION=1.5
```
Comment (Ian-Foote) [inline .travis.yml]: Time to drop `1.5` support I think.

Comment (smcoll) [inline .travis.yml]: Yeah, that was my first though, although it appears that DRF3 support goes [all the way back to 1.4](https://github.com/tomchristie/django-rest-framework/tree/master#requirements)!  i'm fine either way,

File: setup.py (hunk 1 above)
Comment (smcoll) [inline setup.py]: Yes, you're right; this isn't DRF2.4-compatible.

File: .travis.yml (hunk 2 above)
Comment (Ian-Foote) [inline .travis.yml]: Yeah, but I don't see any point making our tests more complex to support a version of Django that isn't even getting security patches any more.

Comment (smcoll) [inline .travis.yml]: Good point.  The same is true for 1.6 at the moment, and will be true of 1.4 soon.

File: generic_relations/relations.py [hunk 3]
```diff
@@ -30,46 +30,27 @@ def __init__(self, serializers, *args, **kwargs):
...
-        serializer.initialize(self.parent, self.source)
-        return serializer.from_native(value)
+        # Necessary because of context, field resolving etc.
+        serializer.bind(self.field_name, self.parent)
+        return serializer.to_internal_value(data)
+
+    def to_representation(self, instance):
+        serializer = self.determine_deserializer_for_data(instance)
+        # clear source before binding, to avoid an AssertionError
+        serializer.source = None
```
Comment (smcoll) [inline generic_relations/relations.py]: Wanted your eyes on this.  It looks hacky, but without it, DRF's serializer raises [an assertion error](https://github.com/tomchristie/django-rest-framework/blob/3.1.3/rest_framework/fields.py#L227).

Comment (Ian-Foote) [inline generic_relations/relations.py]: Why is `source` set on the `serializer`?

Comment (smcoll) [inline generic_relations/relations.py]: Good question.  i thought i had seen that it was set automatically when the serializer was first initialized based on the field_name value, but i'm not seeing that now.  If we take this line out, though, you'll see the error.

Comment (Ian-Foote) [inline generic_relations/relations.py]: Is `to_internal_value` being called before `to_representation`? In that case we're calling `bind` twice which will set `source` in the first call and fail in the `second`.

Comment (Ian-Foote) [inline generic_relations/relations.py]: If this is the problem, I think instead we should be calling `serializer.bind` in `__init__`.

Comment (wimglenn) [inline generic_relations/relations.py]: See https://github.com/tomchristie/django-rest-framework/issues/2555
Tom Christie says the bind should be considered private API, and best is to perform any behavior like that in `MySerializer.__init__`

Comment (craigds) [inline generic_relations/relations.py]: serializer.bind() actually sets serializer.source to the field_name. So if you call it twice you get this error.

Hence setting it in `__init__` is the best option. `ListSerializer` does that: https://github.com/tomchristie/django-rest-framework/blob/06dd55ac1ca6281af2b41b3ad12cc33a2fde2fce/rest_framework/serializers.py#L530

**Comment by Ian-Foote** [comment]
Thanks @smcoll for the initial work on this. I'm sorry it took this long to merge and release.

**Pull request merged**